src/prefect/utilities/annotations.py

```diff
@@ -1,20 +1,32 @@
+import sys
 import warnings
 from abc import ABC
+from collections import namedtuple
 from typing import Generic, TypeVar
 
 T = TypeVar("T")
 
 
-class BaseAnnotation(ABC, Generic[T]):
-    """
-    Base class for Prefect annotation types
+class BaseAnnotation(
+    namedtuple("BaseAnnotation", field_names="value"), ABC, Generic[T]
+):
     """
+    Base class for Prefect annotation types.
 
-    def __init__(self, value: T):
-        self.value = value
+    Inherits from `namedtuple` for unpacking support in another tools.
+    """
 
     def unwrap(self) -> T:
-        return self.value
+        if sys.version_info < (3, 8):
+            # cannot simply return self.value due to recursion error in Python 3.7
+            # also _asdict does not follow convention; it's not an internal method
+            # https://stackoverflow.com/a/26180604
+            return self._asdict()["value"]
+        else:
+            return self.value
+
+    def rewrap(self, value: T) -> "BaseAnnotation[T]":
+        return type(self)(value)
 
     def rewrap(self, value: T) -> "BaseAnnotation[T]":
         return type(self)(value)
```

Comment:
@ahuang11 looks like this got duplicated

Reply:
Thanks; fixed it.